app/upgrades/v8/upgrades.go

```diff
@@ -49,6 +52,65 @@ func FixBankMetadata(ctx sdk.Context, keepers *keepers.AppKeepers) error {
 	return nil
 }
 
+func QuicksilverFix(ctx sdk.Context, keepers *keepers.AppKeepers) error {
+	// Refund stuck coins from ica address
+	sourceAddress, err := sdk.AccAddressFromBech32("cosmos13dqvh4qtg4gzczuktgnw8gc2ewnwmhdwnctekxctyr4azz4dcyysecgq7e")
+
+	if err != nil {
+		return errors.New("invalid source address")
+	}
+	destinationAddress, err := sdk.AccAddressFromBech32("cosmos1jc24kwznud9m3mwqmcz3xw33ndjuufnghstaag")
+	if err != nil {
+		return errors.New("invalid destination address")
+	}
+
+	// Get balance from stuck address and subtract 1 uatom sent by bad actor
+	sourceBalance := keepers.BankKeeper.GetBalance(ctx, sourceAddress, "uatom")
+	refundBalance := sourceBalance.SubAmount(sdk.NewInt(1))
+	keepers.BankKeeper.SendCoins(ctx, sourceAddress, destinationAddress, sdk.NewCoins(refundBalance))
+
+	// Get connection to quicksilver chain
+	connectionId, err := getConnectionIdForChainId(ctx, keepers, "quicksilver-1")
+	if err != nil {
+		return err
+	}
+
+	// Close channels
+	closeChannel(keepers, ctx, connectionId, "icacontroller-cosmoshub-4.deposit")
+	closeChannel(keepers, ctx, connectionId, "icacontroller-cosmoshub-4.withdrawal")
+	closeChannel(keepers, ctx, connectionId, "icacontroller-cosmoshub-4.performance")
+	closeChannel(keepers, ctx, connectionId, "icacontroller-cosmoshub-4.delegate")
```

Comment:
Another question: why do the channels with counterparty port IDs `icacontroller-cosmoshub-4.deposit`, `icacontroller-cosmoshub-4.withdrawal` and `icacontroller-cosmoshub-4.performance` also need to be closed if their ICA acccounts on the Hub were not affected?

Reply:
We want to return to a state where Quicksilver can re-register the zone from scratch (i.e. channels closed on both ends) so return to the desired state.